src/traffic_layout/engine.cc

```diff
@@ -566,6 +332,99 @@ RunrootEngine::copy_runroot(const std::string &original_root, const std::string
   } else {
     path_map[LAYOUT_PREFIX] = ".";
   }
+
+  YAML::Emitter yamlfile;
+  // emit key value pairs to the yaml file
+  yamlfile << YAML::BeginMap;
+  for (uint i = 0; i < dir_vector.size(); i++) {
+    yamlfile << YAML::Key << dir_vector[i];
+    yamlfile << YAML::Value << path_map[dir_vector[i]];
+  }
+  yamlfile << YAML::EndMap;
+
+  // create the file
+  std::ofstream f(Layout::relative_to(ts_runroot, "runroot.yaml"));
+  f << yamlfile.c_str();
+}
+
+// check if user want to force remove the ts_runroot
+// return true if user replies Y
+static bool
+check_remove_force()
+{
+  // check for Y/N 3 times
+  for (int i = 0; i < 3; i++) {
+    std::cout << "Are you sure to force removing runroot? (irreversible) Y/N: ";
```

Comment:
Provide different message here, to make it clear it's repeatedly deliberately and not due to a bug. E.g.
"Are you sure you want to remove runroot?"
"Are  you really sure you want to forcibly remove runroot?"
"I ask you three times, and three times you must agree - remove runroot?"